redis.go

```diff
@@ -290,27 +290,7 @@ func (c *baseClient) withConn(
 		c.releaseConn(ctx, cn, err)
 	}()
 
-	done := ctx.Done() //nolint:ifshort
-
-	if done == nil {
-		err = fn(ctx, cn)
-		return err
-	}
-
-	errc := make(chan error, 1)
-	go func() { errc <- fn(ctx, cn) }()
-
-	select {
-	case <-done:
-		_ = cn.Close()
-		// Wait for the goroutine to finish and send something.
-		<-errc
-
-		err = ctx.Err()
-		return err
-	case err = <-errc:
-		return err
-	}
+	return fn(ctx, cn)
```

Comment:
Please note that this is a rather big change compared to `v8`, and it should be documented as part of the changelog. 

Compared to `v8`, it is no longer possible to use `context.WithCancel` due to this change. I've followed the attempts over time to make `net.Conn` work with context cancellations and I know this is a huge headache, but please at least mention next to the `ContextTimeoutEnabled` entry in the changelog that asynchronous cancellation is no longer supported in `v9`.

Reply:
Done in https://github.com/go-redis/redis/commit/4cb9d14eb6b7f60f967e7f2c040c5795a847e0a0